azure-function/index.js

```diff
@@ -109,12 +109,17 @@ module.exports = async (context, req) => {
 
             const comment = req.body.comment;
             const pullRequestURL = req.body.issue.pull_request.html_url;
-            const prNumber = pullRequestURL.match(/https:\/\/github\.com\/gitgitgadget\/git\/pull\/([0-9]*)$/)
-            if (!comment.id || !prNumber) {
+            const match = pullRequestURL.match(/https:\/\/github\.com\/([^/]*)\/git\/pull\/([0-9]*)$/);
+            if (!comment || !comment.id || !match || match[1] != req.body.repository.owner.login) {
```

Comment:
Why not use `req.body.issue.pull_request.number` and `req.body.repository.owner.login` instead of extracting from the url?

Reply:
I remember now why I did not do it that way: it does not work. You might _think_ that `req.body.issue.pull_request` is a fully-populated JSON as e.g. described at https://developer.github.com/v3/pulls/#get-a-single-pull-request, but it is not. The full `req.body` from an example run:

```json
{
  action: 'edited',
  changes: { body: { from: '/test again' } },
  issue: 
   {
     url: 'https://api.github.com/repos/dscho/git/issues/12',
     repository_url: 'https://api.github.com/repos/dscho/git',
     labels_url: 'https://api.github.com/repos/dscho/git/issues/12/labels{/name}',
     comments_url: 'https://api.github.com/repos/dscho/git/issues/12/comments',
     events_url: 'https://api.github.com/repos/dscho/git/issues/12/events',
     html_url: 'https://github.com/dscho/git/pull/12',
     id: 524054282,
     node_id: 'MDExOlB1bGxSZXF1ZXN0MzQxOTAyNDYw',
     number: 12,
     title: '[TEST] rebase -r: let `label` generate safer labels',
     user: 
      {
        login: 'dscho',
        id: 127790,
        node_id: 'MDQ6VXNlcjEyNzc5MA==',
        avatar_url: 'https://avatars0.githubusercontent.com/u/127790?v=4',
        gravatar_id: '',
        url: 'https://api.github.com/users/dscho',
        html_url: 'https://github.com/dscho',
        followers_url: 'https://api.github.com/users/dscho/followers',
        following_url: 'https://api.github.com/users/dscho/following{/other_user}',
        gists_url: 'https://api.github.com/users/dscho/gists{/gist_id}',
        starred_url: 'https://api.github.com/users/dscho/starred{/owner}{/repo}',
        subscriptions_url: 'https://api.github.com/users/dscho/subscriptions',
        organizations_url: 'https://api.github.com/users/dscho/orgs',
        repos_url: 'https://api.github.com/users/dscho/repos',
        events_url: 'https://api.github.com/users/dscho/events{/privacy}',
        received_events_url: 'https://api.github.com/users/dscho/received_events',
        type: 'User',
        site_admin: false },
     labels: [],
     state: 'open',
     locked: false,
     assignee: null,
     assignees: [],
     milestone: null,
     comments: 13,
     created_at: '2019-11-17T21:57:46Z',
     updated_at: '2019-11-19T16:30:14Z',
     closed_at: null,
     author_association: 'OWNER',
     pull_request: 
      {
        url: 'https://api.github.com/repos/dscho/git/pulls/12',
        html_url: 'https://github.com/dscho/git/pull/12',
        diff_url: 'https://github.com/dscho/git/pull/12.diff',
        patch_url: 'https://github.com/dscho/git/pull/12.patch' },
     body: 'This is a test for GitGitGadget.' },
  comment: 
   {
     url: 'https://api.github.com/repos/dscho/git/issues/comments/555142445',
     html_url: 'https://github.com/dscho/git/pull/12#issuecomment-555142445',
     issue_url: 'https://api.github.com/repos/dscho/git/issues/12',
     id: 555142445,
     node_id: 'MDEyOklzc3VlQ29tbWVudDU1NTE0MjQ0NQ==',
     user: 
      {
        login: 'dscho',
        id: 127790,
        node_id: 'MDQ6VXNlcjEyNzc5MA==',
        avatar_url: 'https://avatars0.githubusercontent.com/u/127790?v=4',
        gravatar_id: '',
        url: 'https://api.github.com/users/dscho',
        html_url: 'https://github.com/dscho',
        followers_url: 'https://api.github.com/users/dscho/followers',
        following_url: 'https://api.github.com/users/dscho/following{/other_user}',
        gists_url: 'https://api.github.com/users/dscho/gists{/gist_id}',
        starred_url: 'https://api.github.com/users/dscho/starred{/owner}{/repo}',
        subscriptions_url: 'https://api.github.com/users/dscho/subscriptions',
        organizations_url: 'https://api.github.com/users/dscho/orgs',
        repos_url: 'https://api.github.com/users/dscho/repos',
        events_url: 'https://api.github.com/users/dscho/events{/privacy}',
        received_events_url: 'https://api.github.com/users/dscho/received_events',
        type: 'User',
        site_admin: false },
     created_at: '2019-11-18T18:18:38Z',
     updated_at: '2019-11-19T16:30:14Z',
     author_association: 'OWNER',
     body: '/testtest again' },
  repository: 
   {
     id: 2326047,
     node_id: 'MDEwOlJlcG9zaXRvcnkyMzI2MDQ3',
     name: 'git',
     full_name: 'dscho/git',
     private: false,
     owner: 
      {
        login: 'dscho',
        id: 127790,
        node_id: 'MDQ6VXNlcjEyNzc5MA==',
        avatar_url: 'https://avatars0.githubusercontent.com/u/127790?v=4',
        gravatar_id: '',
        url: 'https://api.github.com/users/dscho',
        html_url: 'https://github.com/dscho',
        followers_url: 'https://api.github.com/users/dscho/followers',
        following_url: 'https://api.github.com/users/dscho/following{/other_user}',
        gists_url: 'https://api.github.com/users/dscho/gists{/gist_id}',
        starred_url: 'https://api.github.com/users/dscho/starred{/owner}{/repo}',
        subscriptions_url: 'https://api.github.com/users/dscho/subscriptions',
        organizations_url: 'https://api.github.com/users/dscho/orgs',
        repos_url: 'https://api.github.com/users/dscho/repos',
        events_url: 'https://api.github.com/users/dscho/events{/privacy}',
        received_events_url: 'https://api.github.com/users/dscho/received_events',
        type: 'User',
        site_admin: false },
     html_url: 'https://github.com/dscho/git',
     description: 'Git Source Code Mirror',
     fork: true,
     url: 'https://api.github.com/repos/dscho/git',
     forks_url: 'https://api.github.com/repos/dscho/git/forks',
     keys_url: 'https://api.github.com/repos/dscho/git/keys{/key_id}',
     collaborators_url: 'https://api.github.com/repos/dscho/git/collaborators{/collaborator}',
     teams_url: 'https://api.github.com/repos/dscho/git/teams',
     hooks_url: 'https://api.github.com/repos/dscho/git/hooks',
     issue_events_url: 'https://api.github.com/repos/dscho/git/issues/events{/number}',
     events_url: 'https://api.github.com/repos/dscho/git/events',
     assignees_url: 'https://api.github.com/repos/dscho/git/assignees{/user}',
     branches_url: 'https://api.github.com/repos/dscho/git/branches{/branch}',
     tags_url: 'https://api.github.com/repos/dscho/git/tags',
     blobs_url: 'https://api.github.com/repos/dscho/git/git/blobs{/sha}',
     git_tags_url: 'https://api.github.com/repos/dscho/git/git/tags{/sha}',
     git_refs_url: 'https://api.github.com/repos/dscho/git/git/refs{/sha}',
     trees_url: 'https://api.github.com/repos/dscho/git/git/trees{/sha}',
     statuses_url: 'https://api.github.com/repos/dscho/git/statuses/{sha}',
     languages_url: 'https://api.github.com/repos/dscho/git/languages',
     stargazers_url: 'https://api.github.com/repos/dscho/git/stargazers',
     contributors_url: 'https://api.github.com/repos/dscho/git/contributors',
     subscribers_url: 'https://api.github.com/repos/dscho/git/subscribers',
     subscription_url: 'https://api.github.com/repos/dscho/git/subscription',
     commits_url: 'https://api.github.com/repos/dscho/git/commits{/sha}',
     git_commits_url: 'https://api.github.com/repos/dscho/git/git/commits{/sha}',
     comments_url: 'https://api.github.com/repos/dscho/git/comments{/number}',
     issue_comment_url: 'https://api.github.com/repos/dscho/git/issues/comments{/number}',
     contents_url: 'https://api.github.com/repos/dscho/git/contents/{+path}',
     compare_url: 'https://api.github.com/repos/dscho/git/compare/{base}...{head}',
     merges_url: 'https://api.github.com/repos/dscho/git/merges',
     archive_url: 'https://api.github.com/repos/dscho/git/{archive_format}{/ref}',
     downloads_url: 'https://api.github.com/repos/dscho/git/downloads',
     issues_url: 'https://api.github.com/repos/dscho/git/issues{/number}',
     pulls_url: 'https://api.github.com/repos/dscho/git/pulls{/number}',
     milestones_url: 'https://api.github.com/repos/dscho/git/milestones{/number}',
     notifications_url: 'https://api.github.com/repos/dscho/git/notifications{?since,all,participating}',
     labels_url: 'https://api.github.com/repos/dscho/git/labels{/name}',
     releases_url: 'https://api.github.com/repos/dscho/git/releases{/id}',
     deployments_url: 'https://api.github.com/repos/dscho/git/deployments',
     created_at: '2011-09-05T03:18:21Z',
     updated_at: '2019-09-16T20:37:39Z',
     pushed_at: '2019-11-18T22:02:18Z',
     git_url: 'git://github.com/dscho/git.git',
     ssh_url: 'git@github.com:dscho/git.git',
     clone_url: 'https://github.com/dscho/git.git',
     svn_url: 'https://github.com/dscho/git',
     homepage: 'http://git-scm.com',
     size: 192197,
     stargazers_count: 3,
     watchers_count: 3,
     language: 'C',
     has_issues: true,
     has_projects: true,
     has_downloads: true,
     has_wiki: true,
     has_pages: true,
     forks_count: 3,
     mirror_url: null,
     archived: false,
     disabled: false,
     open_issues_count: 1,
     license: 
      {
        key: 'other',
        name: 'Other',
        spdx_id: 'NOASSERTION',
        url: null,
        node_id: 'MDc6TGljZW5zZTA=' },
     forks: 3,
     open_issues: 1,
     watchers: 3,
     default_branch: 'master' },
  sender: 
   {
     login: 'dscho',
     id: 127790,
     node_id: 'MDQ6VXNlcjEyNzc5MA==',
     avatar_url: 'https://avatars0.githubusercontent.com/u/127790?v=4',
     gravatar_id: '',
     url: 'https://api.github.com/users/dscho',
     html_url: 'https://github.com/dscho',
     followers_url: 'https://api.github.com/users/dscho/followers',
     following_url: 'https://api.github.com/users/dscho/following{/other_user}',
     gists_url: 'https://api.github.com/users/dscho/gists{/gist_id}',
     starred_url: 'https://api.github.com/users/dscho/starred{/owner}{/repo}',
     subscriptions_url: 'https://api.github.com/users/dscho/subscriptions',
     organizations_url: 'https://api.github.com/users/dscho/orgs',
     repos_url: 'https://api.github.com/users/dscho/repos',
     events_url: 'https://api.github.com/users/dscho/events{/privacy}',
     received_events_url: 'https://api.github.com/users/dscho/received_events',
     type: 'User',
     site_admin: false
   },
  installation: 
   {
     id: 5187531,
     node_id: 'MDIzOkludGVncmF0aW9uSW5zdGFsbGF0aW9uNTE4NzUzMQ=='
   }
}
```
But just as I stare at this, I realize that I was stupid, and it is the `req.body.issue.number` I was looking for...